src/main/java/org/jabref/logic/exporter/EndnoteXmlExporter.java

```diff
@@ -0,0 +1,209 @@
+
+    private static final String ENDNOTE_XML_VERSION = "20.1";
+    private static final Map<EntryType, String> EXPORT_ITEM_TYPE = Map.ofEntries(
+            Map.entry(StandardEntryType.Article, "Journal Article"),
+            Map.entry(StandardEntryType.Book, "Book"),
+            Map.entry(StandardEntryType.InBook, "Book Section"),
+            Map.entry(StandardEntryType.InCollection, "Book Section"),
+            Map.entry(StandardEntryType.InProceedings, "Conference Paper"),
+            Map.entry(StandardEntryType.Conference, "Conference"),
+            Map.entry(StandardEntryType.MastersThesis, "Thesis"),
+            Map.entry(StandardEntryType.PhdThesis, "Thesis"),
+            Map.entry(StandardEntryType.Proceedings, "Conference Proceedings"),
+            Map.entry(StandardEntryType.TechReport, "Report"),
+            Map.entry(StandardEntryType.Unpublished, "Manuscript"),
+            Map.entry(IEEETranEntryType.Patent, "Patent"),
+            Map.entry(StandardEntryType.Online, "Web Page"),
+            Map.entry(IEEETranEntryType.Electronic, "Electronic Article"),
+            Map.entry(StandardEntryType.Misc, "Generic")
+    );
+
+    private static final Map<EntryType, String> EXPORT_REF_NUMBER = EXPORT_ITEM_TYPE.entrySet().stream()
+                                                                                    .collect(Collectors.toMap(Map.Entry::getKey, entry -> Integer.toString(EXPORT_ITEM_TYPE.entrySet().stream()
+                                                                                                                                                                           .map(Map.Entry::getValue)
+                                                                                                                                                                           .collect(Collectors.toList())
+                                                                                                                                                                           .indexOf(entry.getValue()) + 1)));
+
+    private final XMLOutputFactory xmlOutputFactory;
+    private final AuthorListParser authorListParser;
+
+    public EndnoteXmlExporter() {
+        super("endnote", "EndNote XML", StandardFileType.XML);
+        this.xmlOutputFactory = XMLOutputFactory.newFactory();
+        this.authorListParser = new AuthorListParser();
+    }
+
+    @Override
+    public void export(BibDatabaseContext databaseContext, Path file, List<BibEntry> entries) throws Exception {
+        Objects.requireNonNull(databaseContext);
+        Objects.requireNonNull(file);
+        Objects.requireNonNull(entries);
+
+        if (entries.isEmpty()) {
+            return;
+        }
+
+        try (BufferedWriter writer = Files.newBufferedWriter(file, StandardCharsets.UTF_8)) {
+            XMLStreamWriter xml = xmlOutputFactory.createXMLStreamWriter(writer);
+            xml.writeStartDocument(StandardCharsets.UTF_8.name(), "1.0");
```

Comment:
This would output XML in one line, it's more storage-efficient but it would save you a few KBs or MBs. It's better to output formatted XML for easier debugging.

![image](https://github.com/JabRef/jabref/assets/21198231/63fe3149-12df-4e44-a63d-93e5fe304f0b)

Reply:
Cool, I updated to get formatted XML:
![image](https://github.com/JabRef/jabref/assets/74734844/ed1764bc-f706-476e-9cd5-714ec8e148fa)